TrackletAlgorithm/MemoryTemplateBinnedCM.h

```diff
@@ -38,16 +38,37 @@ class MemoryTemplateBinnedCM{
   
  public:
 
+  MemoryTemplateBinnedCM() {
+    clear();
+  }
+  
+  ~MemoryTemplateBinnedCM() {}
+  
+  void clear() {
+#pragma HLS inline
+	
+  clearloop1:for (size_t ibx=0; ibx<(kNBxBins); ++ibx) {
+#pragma HLS UNROLL
+      clear(ibx);
+    }
+  }
+
+  void clear(BunchXingT bx) {
+#pragma HLS inline
+	
+  clearloop2:for (unsigned int ibin = 0; ibin < 8; ++ibin) {
+#pragma HLS UNROLL
+      nentries8_[bx][ibin] = 0;
+      binmask8_[bx][ibin] = 0;
+    }
+  }
+
   unsigned int getDepth() const {return kNMemDepth;}
   unsigned int getNBX() const {return kNBxBins;}
   unsigned int getNBins() const {return kNSlots;}
-  unsigned int getNEntryPerBin() const {return (1<<(NBIT_ADDR-NBIT_BIN));}
-  unsigned int getNCopy() const {return NCOPY;}
```

Comment:
@bryates One of the reasons the CI tests are failing is because you removed this method. In fact, it looks like you undid many of the changes @meisonlikesicecream made to this file in PR #150. Could you please make sure all your changes are intentional?